src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -217,119 +210,116 @@ def _entity_ids_by_type(self,
         return entity_ids
 
 
+JSON_FILE = TypeVar('JSON_FILE', bound=JsonFile)
+
+
 @attr.s(frozen=True, kw_only=True, auto_attribs=True)
-class GitHubStagingAreaFactory:
-    repo: Repository
-    ref: str
-    path: Tuple[str, ...]
+class CannedStagingAreaFactory:
+    #: Path to a local directory containing one or more staging areas
+    base_path: Path
 
     @classmethod
-    def create(cls,
-               owner: str,
-               name: str,
-               ref: str,
-               path: Optional[str] = None
-               ) -> 'GitHubStagingAreaFactory':
+    def clone_remote(cls, remote_url: furl, local_path: Path, ref: str) -> Self:
         """
-        :param owner: The owner of the GitHub repository
-        :param name: The name of the GitHub repository
-        :param ref: A branch name, tag, or commit SHA
-        :param path: The path inside the repository to the base of the staging area
+        Clone a remote Git repository.
+
+        :param remote_url: The URL of a remote Git repository containing one or
+                           more staging areas
+
+        :param local_path: The path to an empty local directory where the
+                           repository will be cloned
+
+        :param ref: A Git ref (branch, tag, or commit SHA)
+
+        :return: A CannedStagingAreaFactory object configured with the cloned
+                 repository as the base path for creating StagingArea objects
```

Comment:
```suggestion
```

Avoid repeating the name of an artifact in the docstring for that artifact so that future renames don't break the docstring. In this case, this is made redundant by my edit to the docstring title and the fact that the return type is already denoted in the signature.